### PR TITLE
Fix skip [ExcludefromCoverage] for generated async state machine

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Contributes to https://github.com/coverlet-coverage/coverlet/issues/809
We need also to fix issue with lamda.

NB. Cannot add tests because in all other case the `ordinal` value calculation is correct so I cannot find an example where old alg(synthetized names) fails.

cc: @matteoerigozzi @VenkateshSrini

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
Local manual test with repro

image

Still present for lambda

image

File: src/coverlet.core/Instrumentation/Instrumenter.cs [hunk 1]
```diff
@@ -393,6 +424,11 @@ private void InstrumentType(TypeDefinition type)
                         customAttributes = customAttributes.Union(prop.CustomAttributes);
                 }
 
+                if (IsMethodOfCompilerGeneratedClassOfAsyncStateMachineToBeExcluded(method))
+                {
+                    continue;
+                }
+
                 ordinal++;
```
Comment (matteoerigozzi) [inline src/coverlet.core/Instrumentation/Instrumenter.cs:432]: I think this instruction must be moved before the if

Comment (MarcoRossignoli) [reply src/coverlet.core/Instrumentation/Instrumenter.cs]: Done can you take a look?

**Comment by MarcoRossignoli** [comment]
@matteoerigozzi PTAL

**Comment by MarcoRossignoli** [comment]
Test case with lambda bug
image

**Pull request merged**